### PR TITLE
Add AoP docs to SN Hub.

### PR DESCRIPTION
**Pull request by Huanli-Meng**
These items to be updated later, please ignore these three parts:

- source link:
- download link:
- version 

Please help review and it will be appreciated if you can feedback by the end of this Friday (2020.06.05).

### CODE REVIEW

**Comment by Huanli-Meng** [comment]
@gaoran10 , could you please help review the document and provide the following information?
source link:
download link:
version

**Comment by gaoran10** [comment]
@Huanli-Meng Ok, I'll provide the source link, download link and version.

File: protocol-handlers/aop/0.0.1/aop.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+support_img: "/images/connectors/streamnative.png"
+owner_name: "StreamNative"
+owner_img: "/images/streamnative.png"
+dockerfile: 
+id: "aop"
+---
+
+# Overview
+
+AoP (AMQP on Pulsar) supports AMQP0-9-1 protocol and it is backed by Pulsar. AoP is implemented as a Pulsar [protocol handler](https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/protocol/ProtocolHandler.java) with the protocol name of "amqp". The AoP protocol handler is built as a `nar` file and it is loaded when the Pulsar broker starts.
```
Comment (gaoran10) [inline protocol-handlers/aop/0.0.1/aop.md:26]: ProtocolHandler

Comment (gaoran10) [inline protocol-handlers/aop/0.0.1/aop.md:26]:
```suggestion
AoP (AMQP on Pulsar) supports AMQP0-9-1 protocol and it is backed by Pulsar. AoP is implemented as a Pulsar [ProtocolHandler](https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/protocol/ProtocolHandler.java) with the protocol name of "amqp". The AoP protocol handler is built as a `nar` file and it is loaded when the Pulsar broker starts.
```

File: protocol-handlers/aop/0.0.1/aop.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+3. Load the AoP protocol handler. After installing the AoP protocol handler on the Pulsar broker, you can restart the Pulsar broker to load the AoP protocol handler.
+
+# Configuration 
+
+The following table lists configurations available for the AoP protocol handler.
+
+| Item | Description | Default value |
+|---|---|---|
+| amqpTenant | Set the AMQP on the Pulsar broker tenant. | public|
```
Comment (gaoran10) [inline protocol-handlers/aop/0.0.1/aop.md:74]:
```suggestion
| amqpTenant | Set the AMQP on the Pulsar broker tenant. | public |
```

File: protocol-handlers/aop/0.0.1/aop.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+       ```
+       amqpListeners=amqp://127.0.0.1:5672
+       advertisedAddress=127.0.0.1
+       ```
+
+3. Load the AoP protocol handler. After installing the AoP protocol handler on the Pulsar broker, you can restart the Pulsar broker to load the AoP protocol handler.
+
+    ```
+    $PULSAR_HOME/bin/pulsar standalone
```
Comment (gaoran10) [inline protocol-handlers/aop/0.0.1/aop.md:123]:
```suggestion
    $PULSAR_HOME/bin/pulsar-daemon start standalone
```

File: protocol-handlers/aop/0.0.1/aop.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    ```
+
+4. Add the namespace for vhost.
+
+    ```
+    # for example, the vhost name is `vhost`
+    bin/pulsar-admin namespaces create public/vhost1
```
Comment (gaoran10) [inline protocol-handlers/aop/0.0.1/aop.md:130]:
```suggestion
    bin/pulsar-admin namespaces create public/vhost1 -b 1
```

File: protocol-handlers/aop/0.0.1/aop.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+| Item | Description | Default value |
+|---|---|---|
+| amqpTenant | Set the AMQP on the Pulsar broker tenant. | public |
+| amqpListeners | Set the AMQP service port. | amqp://127.0.0.1:5672 |
+| maxNoOfChannels | Set the maximum number of channels available concurrently on a connection. | 64 |
+| maxFrameSize | Set the maximum frame size on a connection. | 4 MB |
+| heartBeat | Set the default heartbeat timeout on the Pulsar broker. | 60 s |
+| amqpProxyPort | Set the AMQP proxy service port. | Port 5682 |
+| useProxy | Enable/Disable the proxy service. | false |
```
Comment (gaoran10) [inline protocol-handlers/aop/0.0.1/aop.md:80]:
```suggestion
| amqpProxyEnable | Enable/Disable the proxy service. | false |
```

File: protocol-handlers/aop/0.0.1/aop.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+language: Java
+document: 
+source: ""
+license: Apache License 2.0
+license_link: "https://www.apache.org/licenses/LICENSE-2.0"
+tags: ["Pulsar", "AMQP", "Protocol handler"]
+alias: AoP
+features: ["AoP brings the native AMQP protocol support to Pulsar by introducing an AMQP protocol handler on Pulsar brokers"]
+icon: "/images/protocol-handler/amqp-logo.png"
+download: ""
```
Comment (gaoran10) [inline protocol-handlers/aop/0.0.1/aop.md:14]:
```suggestion
download: "https://github.com/streamnative/aop/releases/download/v0.1.0/pulsar-protocol-handler-amqp-0.1.0-SNAPSHOT.nar"
```

File: protocol-handlers/aop/0.0.1/aop.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
+---
+description: AoP brings the native AMQP protocol support to Pulsar by introducing an AMQP protocol handler on Pulsar brokers.
+author: ["StreamNative"]
+contributors: ["StreamNative"]
+language: Java
+document: 
+source: ""
```
Comment (gaoran10) [inline protocol-handlers/aop/0.0.1/aop.md:7]:
```suggestion
source: "https://github.com/streamnative/aop/tree/v0.1.0"
```

File: protocol-handlers/aop/0.0.1/aop.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+license: Apache License 2.0
+license_link: "https://www.apache.org/licenses/LICENSE-2.0"
+tags: ["Pulsar", "AMQP", "Protocol handler"]
```
Comment (gaoran10) [inline protocol-handlers/aop/0.0.1/aop.md:10]:
```suggestion
tags: ["Pulsar", "AMQP", "AoP", "Protocol handler"]
```

File: protocol-handlers/aop/0.1.0/aop.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+2. Install the AoP protocol handler.
+
+   1. Set the configuration of the AoP protocol handler.
+       
+       Add the following properties and set their values in the Pulsar configuration file, such as `conf/broker.conf` or `conf/standalone.conf`.
+
+       Property | Recommended value | Default value
+       |---|---|---
+       `messagingProtocols` | amqp | null
+       `protocolHandlerDirectory`| Location of AoP NAR file | ./protocols
```
Comment (gaoran10) [inline protocol-handlers/aop/0.1.0/aop.md:43]:
```suggestion
       `protocolHandlerDirectory`| ./protocols | null
```

File: protocol-handlers/aop/0.1.0/aop.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+
+# Configuration 
+
+The following table lists configurations available for the AoP protocol handler.
+
+| Item | Description | Default value |
+|---|---|---|
+| amqpTenant | Set the AMQP on the Pulsar broker tenant. | public |
+| amqpListeners | Set the AMQP service port. | amqp://127.0.0.1:5672 |
+| maxNoOfChannels | Set the maximum number of channels available concurrently on a connection. | 64 |
```
Comment (gaoran10) [inline protocol-handlers/aop/0.1.0/aop.md:76]:
```suggestion
| amqpMaxNoOfChannels | Set the maximum number of channels available concurrently on a connection. | 64 |
```

File: protocol-handlers/aop/0.1.0/aop.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+| maxFrameSize | Set the maximum frame size on a connection. | 4 MB |
```
Comment (gaoran10) [inline protocol-handlers/aop/0.1.0/aop.md:77]:
```suggestion
| amqpMaxFrameSize | Set the maximum frame size on a connection. | 4194304 (4MB) |
```

File: protocol-handlers/aop/0.1.0/aop.md [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+| heartBeat | Set the default heartbeat timeout on the Pulsar broker. | 60 s |
```
Comment (gaoran10) [inline protocol-handlers/aop/0.1.0/aop.md:78]:
```suggestion
| amqpHeartBeat | Set the default heartbeat timeout on the Pulsar broker. | 60 (s) |
```

File: protocol-handlers/aop/0.1.0/aop.md [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+| amqpProxyPort | Set the AMQP proxy service port. | Port 5682 |
```
Comment (gaoran10) [inline protocol-handlers/aop/0.1.0/aop.md:79]:
```suggestion
| amqpProxyPort | Set the AMQP proxy service port. | 5682 |
```

File: protocol-handlers/aop/0.1.0/aop.md [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+
+    ```
+    $PULSAR_HOME/bin/pulsar-daemon start standalone
+    ```
+
+4. Add the namespace for vhost.
+
+    ```
+    # for example, the vhost name is `vhost`
+    bin/pulsar-admin namespaces create public/vhost1 -b 1
```
Comment (gaoran10) [inline protocol-handlers/aop/0.1.0/aop.md:130]:
```suggestion
    bin/pulsar-admin namespaces create -b 1 public/vhost1
```

**Pull request merged**